src/driver/postgres/PostgresQueryRunner.ts

```diff
@@ -711,31 +711,32 @@ export class PostgresQueryRunner extends BaseQueryRunner implements QueryRunner
-                && !OrmUtils.isArraysEqual(newColumn.enum!, oldColumn.enum!)
+                && (!OrmUtils.isArraysEqual(newColumn.enum!, oldColumn.enum!)
+                    || oldColumn.enumName !== newColumn.enumName)
             ) {
-                const enumName = this.buildEnumName(table, newColumn);
                 const arraySuffix = newColumn.isArray ? "[]" : "";
-                const oldEnumName = this.buildEnumName(table, newColumn, true, false, true);
-                const oldEnumNameWithoutSchema = this.buildEnumName(table, newColumn, false, false, true);
                 const enumTypeBeforeColumnChange = await this.getEnumTypeName(table, oldColumn);
-
-                // rename old ENUM
-                upQueries.push(new Query(`ALTER TYPE "${enumTypeBeforeColumnChange.enumTypeSchema}"."${enumTypeBeforeColumnChange.enumTypeName}" RENAME TO ${oldEnumNameWithoutSchema}`));
-                downQueries.push(new Query(`ALTER TYPE ${oldEnumName} RENAME TO  "${enumTypeBeforeColumnChange.enumTypeName}"`));
-
-                // create new ENUM
+                const prevEnumNameWithSchema = `"${enumTypeBeforeColumnChange.enumTypeSchema}"."${enumTypeBeforeColumnChange.enumTypeName}"`;
+                const prevEnumNameWithOldSuffix = this.buildEnumName(table, oldColumn, false, false, true);
+                const prevEnumNameWithOldSuffixAndSchema = this.buildEnumName(table, oldColumn, true, false, true);
+                const newEnumNameWithSchema = this.buildEnumName(table, newColumn);
+                const newEnumNameWithOldSuffix = this.buildEnumName(table, newColumn, false, false, true);
+                const newEnumNameWithOldSuffixAndSchema = this.buildEnumName(table, newColumn, true, false, true);
+
+                // create new enum type and rename old enum type
+                upQueries.push(new Query(`ALTER TYPE ${prevEnumNameWithSchema} RENAME TO ${prevEnumNameWithOldSuffix}`));
                 upQueries.push(this.createEnumTypeSql(table, newColumn));
-                downQueries.push(this.dropEnumTypeSql(table, oldColumn));
+                downQueries.push(this.dropEnumTypeSql(table, newColumn, newEnumNameWithOldSuffixAndSchema));
 
-                // if column have default value, we must drop it to avoid issues with type casting
+                // if column has default value, we must drop it to avoid issues with type casting
                 if (newColumn.default !== null && newColumn.default !== undefined) {
                     upQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" DROP DEFAULT`));
-                    downQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" SET DEFAULT ${newColumn.default}`));
+                    downQueries.push(new Query(`ALTER TABLE ${this.escapePath(table)} ALTER COLUMN "${newColumn.name}" SET DEFAULT ${oldColumn.default}`));
```

Comment:
Is this correct?  This is setting to the old column def default

Reply:
It's counterintuitive but I believe it's correct - lines 733 - 749 are a block together, if you read the up queries in order from top down and the down queries from bottom up, they match one another i believe - its been a while since i've looked at this but i believe what's going on is applying the `oldColumn.default` value restores whatever was previously the default. Though perhaps we should be only doing this logic if the old column had a default at all, rather than always. a test for this would probably also be helpful!